src/components/Checkbox/Checkbox.tsx

```diff
@@ -43,37 +51,25 @@ export type CheckboxProps = Omit<CheckboxInputProps, 'id'> & {
  *
  * NOTE: Requires either a visible label or `aria-label` prop.
  */
-export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
-  (props, ref) => {
-    // All remaining props are passed to the `input` element
-    const { className, id, label, size = 'lg', disabled, ...other } = props;
+export const Checkbox: CheckboxType = React.forwardRef((props, ref) => {
+  // All remaining props are passed to the `input` element
+  const { className, id, label, size = 'lg', disabled, ...other } = props;
 
-    const generatedId = useId();
-    const checkboxId = id || generatedId;
+  const generatedId = useId();
+  const checkboxId = id || generatedId;
 
-    return (
-      <div className={clsx(className, styles.checkbox)}>
-        <CheckboxInput
-          disabled={disabled}
-          id={checkboxId}
-          ref={ref}
-          {...other}
-        />
-        {label && (
-          <CheckboxLabel disabled={disabled} htmlFor={checkboxId} size={size}>
-            {label}
-          </CheckboxLabel>
-        )}
-      </div>
-    );
-  },
-);
+  return (
+    <div className={clsx(className, styles.checkbox)}>
+      <CheckboxInput disabled={disabled} id={checkboxId} ref={ref} {...other} />
+      {label && (
+        <CheckboxLabel disabled={disabled} htmlFor={checkboxId} size={size}>
+          {label}
+        </CheckboxLabel>
+      )}
+    </div>
+  );
+});
 
-/**
- * NOTE: usually, we would re-export subcomponents with
- *   Checkbox.Input = CheckboxInput;
- *   Checkbox.Label = CheckboxLabel;
- * but this does not compile with Typescript since Checkbox itself is a
- * forwarded ref component. Thus Input and Label must be imported separately.
- */
 Checkbox.displayName = 'Checkbox';
+Checkbox.Input = CheckboxInput;
+Checkbox.Label = CheckboxLabel;
```

Comment:
I'm not sure if we will need to export these parts, as they should all be using the already-exported `Label` component. Tho, I suppose we do have a few counter examples...

Reply:
Yea I can revert this one, since `<CheckboxInput>` usage pattern has already been established as well